cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/schedule/trigger/TriggeringInfos.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.runtime.schedule.trigger;
+
+import com.google.gson.Gson;
+import io.cdap.cdap.api.schedule.PartitionTriggerInfo;
+import io.cdap.cdap.api.schedule.ProgramStatusTriggerInfo;
+import io.cdap.cdap.api.schedule.TimeTriggerInfo;
+import io.cdap.cdap.api.schedule.Trigger;
+import io.cdap.cdap.api.schedule.TriggerInfo;
+import io.cdap.cdap.api.schedule.TriggeringScheduleInfo;
+import io.cdap.cdap.proto.ProgramType;
+import io.cdap.cdap.proto.TriggeringInfo;
+import io.cdap.cdap.proto.TriggeringPropertyMapping;
+import io.cdap.cdap.proto.id.ProgramRunId;
+import io.cdap.cdap.proto.id.ScheduleId;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+/**
+ * Util functions for {@link TriggeringInfo}
+ */
+public final class TriggeringInfos {
+
+  private static final Gson gson = new Gson();
+  public static final String TRIGGERING_PROPERTIES_MAPPING = "triggering.properties.mapping";
```

Comment:
I don't understand what is happening here, why are we looking for a specific property? 

Can you help explain why we need all of these classes instead of just re-using whatever already exists?